packages/plugin-e2e/src/models/panels/visualizations.ts

```diff
@@ -0,0 +1,96 @@
+import { Locator } from '@playwright/test';
+import { PluginTestCtx, VisualizationType } from '../../types';
+
+export const visualizationDataExtractorFactory = (visualizationType: VisualizationType) => {
+  switch (visualizationType) {
+    case 'Table':
+      return new Table();
+    case 'Time series':
+      return new TimeSeries();
+    case 'Stat':
+      return new Stat();
+    case 'Gauge':
+      return new Gauge();
+    case 'Bar gauge':
+      return new BarGauge();
+    default:
+      console.log(`No visualization data extractor found for visualization type: ${visualizationType}`);
+      return new VisualizationDataExtractor(visualizationType);
+  }
+};
+
+export class VisualizationDataExtractor {
```

Comment:
Yeah you may be right. I think I mostly wanted to support different visualizations because it would partly enable E2E testing that a provisioned databoard is working as expect. 

```ts
const dashboard = await readProvision<Dashboard>({ filePath: 'dashboards/redshift.json' });
 const dashboardPage = new DashboardPage({ page, selectors, grafanaVersion, request }, { uid: dashboard.uid });
 await dashboardPage.goto();
 // this is a Gauge panel
 const panel = await dashboardPage.getPanelByTitle('Redshift');
 expect(await panel.getValues()). toHaveValue('test')
```

But ofc this would work also by switching to table view.